crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -2623,7 +2626,22 @@ impl<'a, 'db> ArgumentTypeChecker<'a, 'db> {
             .apply_specialization(self.db, isolated_specialization);
 
         let mut try_infer_tcx = || {
-            let return_ty = self.signature.return_ty?;
+            // For generic constructors, we use the type context to infer the specialization of the class
```

Comment:
I was a bit confused here at first, because "generic constructor" sounded to me like the function itself was generic (which is probably illegal? or doesn't make sense because `__init__` always returns `None`, and so every generic parameter could be annotated with its upper bound). Maybe:
```suggestion
            // For constructors of generic classes, we use the type context to infer the specialization of the class
```